ocean/helium/data_transfer.py

```diff
@@ -189,6 +214,17 @@ def upload_file(src_path, dest_path, meta):
         futures = []
         for f in src_file_list:
             real_dest_path = key + str(f.relative_to(src_root)) if (not key or key.endswith('/')) else key
+
+            try:
+                meta_bytes = xattr.getxattr(f, HELIUM_XATTR)
+                meta = json.loads(meta_bytes.decode('utf-8'))
+            except IOError:
+                # No metadata
+                meta = {}
+            except ValueError:
+                raise ValueError("Source path contains invalid metadata")
```

Comment:
should this be 
```
except ValueError as e:
    raise ValueError("Source path contains invalid metadata") from e
```
or are you trying to mask the underlying error message?

Reply:
Just didn't think about it. Though `from e` won't work in Python2, if we decide to actually support it.

Reply:
Actually, I think it's done automatically; we'd need `raise ... from None` to suppress it.